filelog.go

```diff
@@ -136,28 +177,58 @@ func (w *FileLogWriter) Rotate() {
 	w.rot <- true
 }
 
-// If this is called in a threaded context, it MUST be synchronized
-func (w *FileLogWriter) intRotate() error {
-	// Close any log file that may be open
-	if w.file != nil {
-		fmt.Fprint(w.file, FormatLogRecord(w.trailer, &LogRecord{Created: time.Now()}))
-		w.file.Close()
+// Generate the next filename for rotation using integer suffix
+func (w *FileLogWriter) nextIntegerFilename(filename string) (string, error) {
+	for i := 1; i <= 999; i++ {
```

Comment:
We could crank up the number, but eventually you run into a situation where that loop could churn for a long, long time. The pathological case could cause a lot of unnecessary disk IO.